File: cloud-storage-spring-java/src/main/java/com/github/sevtech/cloud/storage/spring/bean/UploadFileRequest.java
package com.github.sevtech.cloud.storage.spring.bean;

import com.amazonaws.services.s3.model.CannedAccessControlList;
import lombok.Builder;
import lombok.Getter;
import lombok.NonNull;

import java.io.InputStream;

@Getter
@Builder
public class UploadFileRequest {
    @NonNull
    private InputStream stream;
    @NonNull
    private String folder;
    @NonNull
    private String name;
    @NonNull
    private String contentType;
    private String bucketName;
    @Builder.Default
    private CannedAccessControlList accessControl = CannedAccessControlList.Private;
}

File: cloud-storage-spring-kotlin/src/main/kotlin/com/github/sevtech/cloud/storage/spring/bean/UploadFileRequest.kt
package com.github.sevtech.cloud.storage.spring.bean

import com.amazonaws.services.s3.model.CannedAccessControlList
import java.io.InputStream

class UploadFileRequest(var stream: InputStream,
                        var folder: String,
                        var name: String,
                        var contentType: String,
                        var bucketName: String? = null,
                        var accessControl: CannedAccessControlList = CannedAccessControlList.Private)